test/Ansatz/Chain_test.jl

```diff
@@ -107,6 +107,36 @@
     @testset "Canonization" begin
         using Tenet
 
+        @testset "contract" begin
+            qtn = rand(Chain, Open, State; n = 5, p = 2, χ = 20)
+            canonized = canonize(qtn)
+
+            @test_throws ArgumentError contract(canonized, :between, Site(1), Site(2); direction = :dummy)
```

Comment:
`contract` calls `contract!`, so it's not needed
```suggestion
```

Reply:
Before that, we had problems with `contract!` but not `contract` due to `valsplit`. Now I don't think it's necessary...